internal/checker/symboltracker.go

```diff
@@ -0,0 +1,140 @@
+package checker
+
+import (
+	"github.com/microsoft/typescript-go/internal/ast"
+	"github.com/microsoft/typescript-go/internal/modulespecifiers"
+	"github.com/microsoft/typescript-go/internal/nodebuilder"
+)
+
+type SymbolTrackerImpl struct {
+	context            *NodeBuilderContext
+	inner              nodebuilder.SymbolTracker
```

Comment:
The number of things that are types that are `Impl` or `Interface` and members that are `impl` or `inner` makes me feel like something is in the wrong place and/or we’re trying to be too clever with polymorphism. How much of this is temporary until more things are moved out of checker.ts?

Reply:
The `SymbolTracker` interface defined outside `checker` *is* required - otherwise the printer and checker end up circularly dependent on one another, so don't work. I did just push a removal of the `NodeBuilderInterface` since, for now at least, it's not strictly required.